main.tf

```diff
@@ -169,7 +169,7 @@ resource "aws_rds_cluster" "this" {
 ################################################################################
 
 resource "aws_rds_cluster_instance" "this" {
-  for_each = { for k, v in var.instances : k => v if local.create_cluster && !local.is_serverless }
+  for_each = { for k, v in var.instances : k => v if local.create_cluster && !local.is_serverless && !var.is_headless_cluster }
```

Comment:
I don't follow this logic - wouldn't a headless cluster just be a cluster without any `instances` defined?